python-checks/src/main/java/org/sonar/python/checks/IsCloseAbsTolCheck.java

```diff
@@ -44,21 +51,35 @@ public void initialize(Context context) {
   }
 
   private static void checkForIsCloseAbsTolArgument(SubscriptionContext ctx, CallExpression call) {
-    Optional.ofNullable(call.calleeSymbol())
-        .filter(symbol -> "math.isclose".equals(symbol.fullyQualifiedName()) &&
-            anyArgumentIsZero(call) &&
-            TreeUtils.argumentByKeyword("abs_tol", call.arguments()) == null)
-        .ifPresent(s -> ctx.addIssue(call, MESSAGE));
+    Symbol symbol = call.calleeSymbol();
+    if (symbol != null && "math.isclose".equals(symbol.fullyQualifiedName())
+        && TreeUtils.argumentByKeyword("abs_tol", call.arguments()) == null) {
+      RegularArgument firstArg = TreeUtils.nthArgumentOrKeyword(0, "a", call.arguments());
+      RegularArgument secondArg = TreeUtils.nthArgumentOrKeyword(1, "b", call.arguments());
+      checkArgumentExistsAndIsZero(firstArg)
+          .ifPresentOrElse(
+              argA -> addIssueAndQuickFix(argA, ctx, call),
+              () -> checkArgumentExistsAndIsZero(secondArg).ifPresent(argB -> addIssueAndQuickFix(argB, ctx, call)));
+    }
   }
 
-  private static boolean anyArgumentIsZero(CallExpression call) {
-    RegularArgument firstArg = TreeUtils.nthArgumentOrKeyword(0, "a", call.arguments());
-    RegularArgument secondArg = TreeUtils.nthArgumentOrKeyword(1, "b", call.arguments());
-    return (firstArg != null && isLitteralZeroOrAssignedZero(firstArg.expression())) ||
-        (secondArg != null && isLitteralZeroOrAssignedZero(secondArg.expression()));
+  private static Optional<RegularArgument> checkArgumentExistsAndIsZero(@Nullable RegularArgument argument) {
+    return Optional
+        .ofNullable(argument)
+        .filter(arg -> isLiteralZeroOrAssignedZero(arg.expression()));
+  }
+
+  private static void addIssueAndQuickFix(RegularArgument arg, SubscriptionContext ctx, CallExpression call) {
```

Comment:
This looks good but it's missing a unit test: we have a `PythonQuickFixVerifier` class that allows you to test before / after snippets of code when applying quick fixes, so that we're sure that everything displays properly.

Reply:
Thank you, yes, I went too fast!